src/FsAutoComplete.Core/Commands.fs

```diff
@@ -1220,21 +1212,19 @@ type Commands (serialize : Serializer, backgroundServiceEnabled, toolsPath) =
           | false, None ->
             currentIndex, false, acc
 
-        let! hints =
+        let hints =
           contents
           |> Array.map (Lexer.tokenizeLine [||])
           |> Array.mapi (fun currentIndex currentTokens -> currentIndex, currentTokens)
           |> Array.fold folder (0, false, [])
-          |> (fun (_, _, third) -> third)
-          |> List.map (fun (lastExpressionLine, lastExpressionLineWasPipe, currentIndex, pipeToken) ->
-              async {
-                let! gens = getGenerics currentIndex pipeToken
-                let previousNonPipeLine = if lastExpressionLineWasPipe then None else Some lastExpressionLine
-                return currentIndex, previousNonPipeLine, gens
-              })
-          |> Async.Parallel
+          |> (fun (_, _, third) -> third |> Array.ofList)
+          |> Array.Parallel.map (fun (lastExpressionLine, lastExpressionLineWasPipe, currentIndex, pipeToken) ->
```

Comment:
this is still done in parallel, just a bit differently since things aren't async anymore.